tests/infrastructure/textual/widgets/test_agent_workspace_logic.py

```diff
@@ -0,0 +1,52 @@
+
+import pytest
+from pathlib import Path
+from simple_agent.infrastructure.textual.widgets.agent_workspace import AgentWorkspace
+from simple_agent.application.agent_id import AgentId
+from textual.widgets import Markdown
+
+def test_refresh_todos_updates_content_from_file(tmp_path):
+    agent_id = AgentId("test_agent")
+    # Use the production logic to determine the filename
+    filename = agent_id.todo_filename()
+    todo_file = tmp_path / filename
+    todo_file.write_text("Initial content", encoding="utf-8")
+
+    # Override todo_filename to return the absolute path to our temp file
+    # This allows us to use tmp_path without changing the process working directory
+    agent_id.todo_filename = lambda: str(todo_file)
```

Comment:
Why do we need this code?
Either we don't need it, or we need todo_filename to always return the absolute path.